docs/js/main.js

```diff
@@ -2070,6 +2070,25 @@ onload = function() {
         }
     });
 
+    document.addEventListener("visibilitychange", function() {
+        if (document.visibilityState === "hidden") {
+            // Save puzzle progress
+            let local_storage_setting = document.getElementById("clear_storage_opt").value;

```

Comment:
Ideally this should check the value of UserSettings.local_storage instead of accessing the form element directly.

Reply:
This makes sense and I changed the code to use `UserSettings` instead. If #157 gets merged I can rebase and use the new settings name.